### ISSUE TITLE
DHCP client is leaking memory.

### ISSUE THREAD

**Issue by andreixc** [tass-belgium/picotcp#13]
This was discovered on the mbed platform.
If you connect the board to a network where there is no DHCP server
for each xid generated and tried for 5 times 100 bytes of ram is lost.
Even if the server responds later on the previous memory is lost.

Each time Ethernet::connect is called it will call pico_dhcp_initiate_negotiation which will create a new cookie, but when it fails there is no way to cleanup the used memory.

Basically if you repeat the call to pico_dhcp_initiate_negotiation memory is leaked.
